### PR TITLE
[Bugfix] #7702 Fix read message, add ability to delete message, cover with tests

### PR DESCRIPTION
**Pull request by yuliiakaras**
[#7702](https://github.com/ita-social-projects/GreenCity/issues/7702)

**Summary of changes:**
1. Added ability to delete notification
2. Fixed ability to read nothification
3. Fixed counter of unread notifications

**New behaviour:**
https://github.com/user-attachments/assets/8f9aa23e-72ef-4851-b07e-d47c3063c9c3

### CODE REVIEW

File: src/app/shared/ubs-base-sidebar/ubs-base-sidebar.component.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export class UbsBaseSidebarComponent implements AfterViewInit, AfterViewChecked,
   }
 
   getIcon(listItem): string {
```
Comment (Lokankara) [inline src/app/shared/ubs-base-sidebar/ubs-base-sidebar.component.ts]: By explicitly defining the type for `listItem`, you enhance type safety and clarity in code:
`getIcon(listItem: listElements): string `

File: src/app/ubs/ubs-user/services/user-messages.service.ts [hunk 2]
```diff
@@ -30,8 +30,12 @@ export class UserMessagesService implements OnDestroy {
     return this.http.get<number>(`${this.url}/notifications/quantityUnreadenNotifications`);
   }
 
-  setReadNotification(id: number): Observable<any> {
-    return this.http.post(`${this.url}/notifications/${id}?lang=${this.language}`, {});
+  setReadNotification(id: number): Observable<void> {
```
Comment (Lokankara) [inline src/app/ubs/ubs-user/services/user-messages.service.ts]: name `setReadNotification` may not accurately reflect its purpose, especially if it specifically marks a notification as read: `updateNotificationStatus`, `markNotificationAsRead` etc

File: src/app/ubs/ubs-user/services/user-messages.service.ts [hunk 3]
```diff
@@ -11,13 +11,13 @@ import { takeUntil } from 'rxjs/operators';
...
-  countOfNoReadeMessages: any;
+  private readonly destroyed$: ReplaySubject<any> = new ReplaySubject<any>(1);
+  countOfNoReadMessages: any;
   language: string;
 
   constructor(
-    private http: HttpClient,
-    private localStorageService: LocalStorageService
+    private readonly http: HttpClient,
+    private readonly localStorageService: LocalStorageService
```
Comment (Lokankara) [inline src/app/ubs/ubs-user/services/user-messages.service.ts]: it seems that the LocalStorageService is only being used to fetch the language parameter for the getNotification method, it makes sense to pass the language as a parameter in UbsUserMessagesComponent:
`getNotification(currentPage: number, size: number, language: string){};`
`this.fetchNotification(language);`

File: src/app/ubs/ubs-user/ubs-user-messages/ubs-user-messages.component.spec.ts [hunk 4]
```diff
@@ -3,8 +3,8 @@ import { ComponentFixture, fakeAsync, TestBed, tick, waitForAsync } from '@angul
 import { TranslateModule } from '@ngx-translate/core';
 import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
 import { MatDialogModule } from '@angular/material/dialog';
-// import { HttpClientTestingModule } from '@angular/common/http/testing';
-import { of } from 'rxjs';
+import { MatSnackBarComponent } from '@global-errors/mat-snack-bar/mat-snack-bar.component';
+import { of, throwError } from 'rxjs';
 import { NotificationBody, Notifications } from '../../ubs-admin/models/ubs-user.model';
```
Comment (Lokankara) [inline src/app/ubs/ubs-user/ubs-user-messages/ubs-user-messages.component.spec.ts]: Import can be shortened: `@ubs/ubs-admin/models/ubs-user.model`

File: src/app/ubs/ubs-user/ubs-user-messages/ubs-user-messages.component.ts [hunk 5]
```diff
@@ -21,11 +22,11 @@ export class UbsUserMessagesComponent implements OnInit, OnDestroy {
   count = 0;
   pageSize = 10;
   isLoadSpinner: boolean;
-  isLoadSmallSpinner: boolean;
   isLoadBar: boolean;
+  hasNextPage: boolean;
   images = [];
   countOfMessages: number;
```
Comment (Lokankara) [inline src/app/ubs/ubs-user/ubs-user-messages/ubs-user-messages.component.ts]: Unused field countOfMessages

File: src/app/ubs/ubs-user/ubs-user-messages/ubs-user-messages.component.ts [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
 import { UserMessagesService } from '../services/user-messages.service';
 import { LocalStorageService } from '@global-service/localstorage/local-storage.service';
+import { MatSnackBarComponent } from '@global-errors/mat-snack-bar/mat-snack-bar.component';
 import { NotificationBody } from '../../ubs-admin/models/ubs-user.model';
```
Comment (Lokankara) [inline src/app/ubs/ubs-user/ubs-user-messages/ubs-user-messages.component.ts]: Import can be shortened

**Review by Lokankara — APPROVED** [review]

**Review by kovalsofiia1 — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Pull request merged**